src/Impostor.Server/Net/State/Game.Outgoing.cs

```diff
@@ -100,5 +120,27 @@ private void WriteWaitForHostMessage(IMessageWriter message, bool clear, IClient
         {
             Message12WaitForHostS2C.Serialize(message, clear, Code, player.Client.Id);
         }
+
+        private ValueTask SendObjectSpawn(InnerNetObject obj, int? targetClientId = null)
+        {
+            using var writer = StartGameData(targetClientId);
+            writer.StartMessage(GameDataTag.SpawnFlag);
+            writer.WritePacked(SpawnableObjectIds[obj.GetType()]);
+            writer.WritePacked(obj.OwnerId);
+            writer.Write((byte)obj.SpawnFlags);
+
+            var components = obj.GetComponentsInChildren<InnerNetObject>();
+            writer.WritePacked(components.Count);
+            foreach (var comp in components)
+            {
+                writer.WritePacked(obj.NetId);
+                writer.StartMessage(1);
+                comp.SerializeAsync(writer, true);
+                writer.EndMessage();
+            }
+
+            writer.EndMessage();
+            return FinishGameDataAsync(writer, targetClientId);
```

Comment:
Returning the task instead async+awaiting here means that the writer will get disposed after this method returns and before it's awaited.

Reply:
The same issue is in `OnQueryPlatformIds` :P

Reply:
Fixed, but I'd like to see an static analyzer chek for stuff like this. Do you know if that exists?

I'd not be surprised if this exists in at least 5 other places